source/tools/testConsistency/dotnetWPF-renderer/app/routes.py

```diff
@@ -0,0 +1,67 @@
+# Copyright (c) Microsoft Corporation. All rights reserved. 
+# Licensed under the MIT License. 
+import os 
+import json 
+from flask import request 
+import subprocess 
+from app import app 
+
+@app.route('/', methods=['POST'])
+def index():
+        """This POST request accepts JSON and returns the card screenshot on .NET WPF
+
+            It accepts JSON data via HTTP POST, creates a new JSON string with the new 
+            fields and sends it to the nodejs app. It then receives back the screenshot 
+            of the card to the client.
+
+            Returns:
+                A JSON string that has fields for "imageData" and "errors". The
+                imageData field contains a base 64 encoded string representing the
+                screenshot of the card.
+        """
+        template = request.form['template']
+        data = request.form['data']
+
+        card = ''
+        if len(data) != 0:
+            card = '{"data": ' + data + ', "template" : ' + template + '}'
+        else: 
+            card = '{"data": "" ' + ', "template" : ' + template + '}'
+
+        card_byte_buffer = bytes(card, 'utf-8')  # The subprocess module requires the input to be a byte stream
+
+
+        res_json = subprocess.run(['WPFRenderer/WPFRenderer/bin/Debug/WPFRenderer.exe'],
+                                    input=card_byte_buffer,
+                                    capture_output=True 
+                                    )
+        img_json_str = (res_json.stdout).decode('utf-8')
+        return img_json_str
+
+@app.route('/test', methods=['POST'])
+def test():
+        """This POST request is primarily for testing purposes. 
+
```

Comment:
Ideally this would be simulated on the test client instead of adding mocks in the actual app code. Either ways though this should be behind a config switch that only your test client sets and is allowed to call and not the whole world

Reply:
I wasn't sure how to simulate the situation where we only get an incomplete result. So my solution was to create a test route that would be call a different driver program. The driver program would function in a similar manner as the actual driver except it would not actually render the card. It would simply send a partial/incomplete base64 encoded string. 
Does that sound reasonable? Let me know if there is a better way to simulate such a situation